src/plugins/kibana/public/visualize/editor/agg_group.js

```diff
@@ -40,6 +42,29 @@ uiModules
           if (count < schema.max) return true;
         });
       });
+
+      const drake = dragula({
+        containers: $('.vis-editor-agg-group', $el).toArray(),
+        moves(el, source, handle) {
+          const $handle = $(handle);
+          return $handle.add($handle.parent()).is('.vis-editor-agg-header-drag-handle');
+        }
+      });
+
+      drake.on('drop', (el, target, source, sibling) => {
+        const aggs = $scope.group;
+        const aggScope = $(el).scope();
+        _.move(aggs, aggScope.agg, getIndex(sibling));
+        function getIndex(el) {
+          if (!el) { return aggs.length - 1; }
+          const list = $('.vis-editor-agg-wrapper', target);
+          return list.index(el) - 1;
+        }
+      });
+      drake.on('cloned', el => $('.vis-editor-agg-add', el).remove());
```

Comment:
removing the add button this way fails when the last element in the list is dragged, moved around, but ultimately not reordered. I would prefer that we hid the button with angular, by setting a property on scope or something when dragging starts/stops (`$scope.dragging = true/false` maybe`), then observe that value in the view